torch/csrc/api/include/torch/nn/functional/linear.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <torch/nn/options/activation.h>
+#include <torch/types.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor linear(const Tensor& input, const Tensor& weight,
+                     const Tensor& bias) {
+  return torch::linear(input, weight, bias);
```

Comment:
I think we should strictly follow the Python implementation here:
```python
    if input.dim() == 2 and bias is not None:
        # fused op is marginally faster
        ret = torch.addmm(bias, input, weight.t())
    else:
        output = input.matmul(weight.t())
        if bias is not None:
            output += bias
        ret = output
    return ret
```
otherwise people would complain about difference in performance between Python and C++ code :(